api/store/mongo/device.go

```diff
@@ -282,7 +282,7 @@ func (s *Store) DeviceDelete(ctx context.Context, uid models.UID) error {
 	return err
 }
 
-func (s *Store) DeviceCreate(ctx context.Context, d models.Device, hostname string) error {
+func (s *Store) DeviceCreate(ctx context.Context, d models.Device, hostname string) (bool, error) {
```

Comment:
I'd like to know if this API signature is the best one. I understand that you are trying to figure out whether it has been inserted or not. Boolean is not a straightforward API because it does not communicate other possible values for the feature. What about using constants that are clear when comparing whether it was or was not inserted?